File: Algorithms/Python3.x/1431-Kids_With_the_Greatest_Number_of_Candies.py
# Runtime: 24 ms, faster than 99.90% of Python3 online submissions for Kids With the Greatest Number of Candies.
# Memory Usage: 13.9 MB, less than 44.78% of Python3 online submissions for Kids With the Greatest Number of Candies.
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        res = []
        max_candies = max(candies)
        for i in candies:
            res.append(i + extraCandies >= max_candies)
        return res

File: Algorithms/C++/1431-Kids_With_the_Greatest_Number_of_Candies.cpp
// Runtime: 4 ms, faster than 86.35% of C++ online submissions for Kids With the Greatest Number of Candies.
// Memory Usage: 9.1 MB, less than 49.77% of C++ online submissions for Kids With the Greatest Number of Candies.
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        vector<bool> res(candies.size());
        int max_candies = *std::max_element(candies.begin(), candies.end());
        for (int i = 0; i < candies.size(); ++i) {
            res[i] = candies[i] + extraCandies >= max_candies;
        }
        return res;
    }
};
